src/inference/src/cpp/remote_tensor.cpp

```diff
@@ -55,6 +60,18 @@ AnyMap RemoteTensor::get_params() const {
     }
 }
 
+void RemoteTensor::copy_to(ov::Tensor& dst) const {
+    OPENVINO_ASSERT(_impl != nullptr, "Tensor was not initialized.");
+    auto remote_tensor_impl = std::dynamic_pointer_cast<ov::IRemoteTensor>(_impl);
+    remote_tensor_impl->copy_to(get_tensor_impl(dst)._ptr);
+}
+
+void RemoteTensor::copy_from(ov::Tensor& src) {
+    OPENVINO_ASSERT(_impl != nullptr, "Tensor was not initialized.");
+    auto remote_tensor_impl = std::dynamic_pointer_cast<ov::IRemoteTensor>(_impl);
+    remote_tensor_impl->copy_from(get_tensor_impl(src)._ptr);
```

Comment:
do we need to assert that `remote_tensor_impl` is not null?

Reply:
Removed